lib/Core/StatsTracker.cpp

```diff
@@ -76,11 +76,16 @@ namespace {
                      cl::init(1.),
 		     cl::desc("Approximate number of seconds between stats writes (default=1.0s)"));
 
-  cl::opt<unsigned> StatsWriteAfterInstructions(
+  cl::opt<std::uint32_t> StatsWriteAfterInstructions(
       "stats-write-after-instructions", cl::init(0),
       cl::desc("Write statistics after each n instructions, 0 to disable "
                "(default=0)"));
 
+  cl::opt<std::uint32_t> CommitEvery(
+      "stats-commit-after", cl::init(0),
+      cl::desc("Commit the statistics every N writes. Setting to 0 commits every write in interval mode (default) or every 1000 writes in write after N instructions"
+               "(default=0)"));
```

Comment:
This needs rephrasing. I would expect
```
StatsWriteAfterInstructions CommitEvery
0 (off)                     0 (off)
0 (off)                     n (error)
n                           0 (write == commit)
n                           m (commit after intervals of n*m writes)

Reply:
That's not how it works.
It's:

```
StatsWriteAfterInstructions        WriteInterval            CommitEvery
0 (off)                                                  n                           0 (commit every n seconds)
n                                                         m                          k (error)
0 (off)                                                  0                           n (off (why would I error?))
n                                                         0                           0 (commit every 1000 instructions)
n                                                         0                           m (commit after m writes)
```

If you don't specify CommitEvery and specify WriteAfterInstruction, your version would  not have sane defaults. If you are using WriteAfterInstructions and commit every write, you will most likely take a significant performance penality, which is not the default you want. I argued for that in the PR description.

Reply:
I didn't specify the default values. Thinking about it, I would disable 0 for CommitEvery:
```
StatsWriteAfterInstructions WriteInterval CommitEvery
0                           0             _ (stats off, warn if CommitEvery set)
_                           _             0 (error, no commit)
0                           n             k (write every n seconds, commit every k writes)
m                           0             k (write every m instrs, commit every k writes)
m                           n             k (write every m instrs OR n seconds, commit every k writes)
```
Default values sth. like: 0/1s/30?

Reply:
No, you need different default for write interval/instructions. Also I imagine most users don't want to deal with commitEvery.

Reply:
I've only proposed a sane default for `CommitEvery` and kept the other ones. Proposing an instruction-based default is hard as this depends on the runtime behaviour of the application. Emit a helpful message, use a different default... I'd prefer the message.

Reply:
I don't think users need to worry about `CommitEvery` unless they really want to. That is they want each instruction logged and committed because KLEE might crash. Emitting and error message is just whitenoise people ignore.

Reply:
But you're mangling up responsibilities. If one sets `WriteInterval` to 100ms, you surely want to increase `CommitEvery`. We should keep it separated and simple.

Reply:
Then remove a 0 from my example.